opaque_keys/models.py

```diff
@@ -0,0 +1,226 @@
+"""
+Useful django models for implementing XBlock infrastructure in django.
+"""
+import logging
+import warnings
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.db.models.lookups import IsNull
+
+from opaque_keys.edx.keys import BlockTypeKey, CourseKey, UsageKey
+
+log = logging.getLogger(__name__)
+
+
+class Creator(object):
+    """
+    A placeholder class that provides a way to set the attribute on the model.
+    """
+    def __init__(self, field):
+        self.field = field
+
+    def __get__(self, obj, type=None):  # pylint: disable=redefined-builtin
+        if obj is None:
+            return self
+        return obj.__dict__[self.field.name]
+
+    def __set__(self, obj, value):
+        obj.__dict__[self.field.name] = self.field.to_python(value)
+
+
+# pylint: disable=missing-docstring,unused-argument
+class CreatorMixin(object):
+    """
+    Mixin class to provide SubfieldBase functionality to django fields.
+    See: https://docs.djangoproject.com/en/1.11/releases/1.8/#subfieldbase
```

Comment:
Please add this to this comment:

DO NOT REUSE THIS CLASS. Provided for backwards compatibility only!